include/rocksdb/env.h

```diff
@@ -364,6 +364,8 @@ class Env {
   // Priority for requesting bytes in rate limiter scheduler
   enum IOPriority { IO_LOW = 0, IO_HIGH = 1, IO_TOTAL = 2 };
 
+  enum IOType { IO_UNCATEGORIZED = 0, IO_FLUSH = 1, IO_COMPACTION = 2 };
```

Comment:
I want to avoid adding IO type to rocksdb code. I think the rust `FileSystemInpector` can obtain the thread-local IO type by itself. To tell flush and compaction apart, we can set the thread-local IO type in the `OnFlushBegin`/`OnCompactionBegin` callbacks.

Reply:
Sounds great, save lots of trouble.

Reply:
@yiwu-arbug But I think `OnCompactionBegin` doesn't respect subcompaction.

Reply:
sad. how about adding an sub-compaction begin event?

Reply:
Also `OnFlushComplete` is tricky, it can run on another thread. But maybe we just make use of both of the `Begin` event.

Reply:
It's doable, I'll leave it for followup PRs.